src/sentry/models/release.py

```diff
@@ -175,48 +130,39 @@ def filter_to_semver(self):
-        version is a version string matching semver format (https://semver.org/). We've
-        slightly extended this format to allow up to 4 integers. EG
-         - sentry@1.2.3.4
-         - sentry@1.2.3.4-alpha
-         - 1.2.3.4
-         - 1.2.3.4-alpha
-         - 1.*
+        Filters released based on a based `SemverFilter` instance.
+        `SemverFilter.version_parts` can contain up to 6 components, which should map
+        to the columns defined in `Release.SEMVER_COLS`. If fewer components are
+        included, then we will exclude later columns from the filter.
+        `SemverFilter.package` is optional, and if included we will filter the `package`
+        column using the provided value.
+        `SemverFilter.operator` should be a Django field filter.
+
+        Typically we build a `SemverFilter` via `sentry.search.events.filter.parse_semver`
         """
-        # TODO: Will refactor this further so that we pass a `SemverFilter` instead of
-        # the version string
-        semver_filter = convert_semver_to_filter(version, operator)
-        if semver_filter:
-            # The version matches semver format, so we can use it for comparison
-            release_filter = Q(organization_id=organization_id)
-            if semver_filter.package:
-                release_filter &= Q(package=semver_filter.package)
-
-            filter_func = Func(
-                *[
-                    Value(part) if isinstance(part, str) else part
-                    for part in semver_filter.version_parts
-                ],
-                function="ROW",
-            )
-            cols = self.model.SEMVER_COLS[: len(semver_filter.version_parts)]
-            return (
-                self.filter(release_filter)
-                .annotate_prerelease_column()
-                .annotate(
-                    semver=Func(
-                        *[F(col) for col in cols], function="ROW", output_field=ArrayField()
-                    )
-                )
-                .filter(**{f"semver__{semver_filter.operator}": filter_func})
+        release_filter = Q(organization_id=organization_id)
+        if semver_filter.package:
+            release_filter &= Q(package=semver_filter.package)
+
+        filter_func = Func(
+            *[
+                Value(part) if isinstance(part, str) else part
+                for part in semver_filter.version_parts
+            ],
+            function="ROW",
+        )
+        cols = self.model.SEMVER_COLS[: len(semver_filter.version_parts)]
+        return (
+            self.filter(release_filter)
+            .annotate_prerelease_column()
+            .annotate(
+                semver=Func(*[F(col) for col in cols], function="ROW", output_field=ArrayField())
             )
-        else:
-            raise InvalidSearchQuery(f"Invalid format for semver query {version}")
+            .filter(**{f"semver__{semver_filter.operator}": filter_func})
```

Comment:
This code hasn't really changed, other than removing the guard on a `None` `semver_filter`